### PR TITLE
feat(dli): support queue scaling policies

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

1. The resource supports queue scaling policies to elastic resource pool.
2. Add constraint descriptions for `elastic_resource_pool` parameters

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

When using the parameter `scaling_policies·, you must ensure that there is a scaling policy with a time period from "00:00" to "24:00". Otherwise, the error message is as follows:
image

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDliQueue_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDliQueue_basic -timeout 360m -parallel 4
=== RUN   TestAccDliQueue_basic
=== PAUSE TestAccDliQueue_basic
=== CONT  TestAccDliQueue_basic
--- PASS: TestAccDliQueue_basic (261.68s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       261.728s

make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDliQueue_scalingPolicies'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDliQueue_scalingPolicies -timeout 360m -parallel 4

=== RUN   TestAccDliQueue_scalingPolicies
=== PAUSE TestAccDliQueue_scalingPolicies
=== CONT  TestAccDliQueue_scalingPolicies
--- PASS: TestAccDliQueue_scalingPolicies (4561.65s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       4561.701s
```

### CODE REVIEW

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDliQueue_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDliQueue_basic -timeout 360m -parallel 4
=== RUN   TestAccDliQueue_basic
=== PAUSE TestAccDliQueue_basic
=== CONT  TestAccDliQueue_basic
--- PASS: TestAccDliQueue_basic (263.63s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       263.712s

make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDliQueue_associateElasticResourcePool'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDliQueue_associateElasticResourcePool -timeout 360m -parallel 4
=== RUN   TestAccDliQueue_associateElasticResourcePool
=== PAUSE TestAccDliQueue_associateElasticResourcePool
=== CONT  TestAccDliQueue_associateElasticResourcePool
--- PASS: TestAccDliQueue_associateElasticResourcePool (4536.13s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       4536.194s

 make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDliQueue_scalingPolicies'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDliQueue_scalingPolicies -timeout 360m -parallel 4
=== RUN   TestAccDliQueue_scalingPolicies
=== PAUSE TestAccDliQueue_scalingPolicies
=== CONT  TestAccDliQueue_scalingPolicies
--- PASS: TestAccDliQueue_scalingPolicies (4561.70s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       4561.741s

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDliQueue_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDliQueue_basic -timeout 360m -parallel 4
=== RUN TestAccDliQueue_basic
=== PAUSE TestAccDliQueue_basic
=== CONT TestAccDliQueue_basic
--- PASS: TestAccDliQueue_basic (263.63s)
PASS
ok github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli 263.712s

make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDliQueue_associateElasticResourcePool'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDliQueue_associateElasticResourcePool -timeout 360m -parallel 4
=== RUN TestAccDliQueue_associateElasticResourcePool
=== PAUSE TestAccDliQueue_associateElasticResourcePool
=== CONT TestAccDliQueue_associateElasticResourcePool
--- PASS: TestAccDliQueue_associateElasticResourcePool (4536.13s)
PASS
ok github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli 4536.194s

make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDliQueue_scalingPolicies'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDliQueue_scalingPolicies -timeout 360m -parallel 4
=== RUN TestAccDliQueue_scalingPolicies
=== PAUSE TestAccDliQueue_scalingPolicies
=== CONT TestAccDliQueue_scalingPolicies
--- PASS: TestAccDliQueue_scalingPolicies (4561.70s)
PASS
ok github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli 4561.741s

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDliQueue_associateElasticResourcePool'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDliQueue_associateElasticResourcePool -timeout 360m -parallel 4
=== RUN   TestAccDliQueue_associateElasticResourcePool
=== PAUSE TestAccDliQueue_associateElasticResourcePool
=== CONT  TestAccDliQueue_associateElasticResourcePool
--- PASS: TestAccDliQueue_associateElasticResourcePool (4539.47s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       4539.527s

 make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDliQueue_scalingPolicies'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDliQueue_scalingPolicies -timeout 360m -parallel 4
=== RUN   TestAccDliQueue_scalingPolicies
=== PAUSE TestAccDliQueue_scalingPolicies
=== CONT  TestAccDliQueue_scalingPolicies
--- PASS: TestAccDliQueue_scalingPolicies (4550.81s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       4550.856s

File: huaweicloud/services/dli/resource_huaweicloud_dli_queue.go [hunk 1]
```diff
@@ -275,6 +318,17 @@ func resourceDliQueueRead(_ context.Context, d *schema.ResourceData, meta interf
...
 
+	if elasticResourcePoolName, ok := d.GetOk("elastic_resource_pool_name"); ok {
+		policies, err := getQueueScalingPolicies(cfg, region, elasticResourcePoolName.(string), queueName)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		mErr = multierror.Append(
+			d.Set("scaling_policies", policies),
+		)
```
Comment (Lance52259) [inline huaweicloud/services/dli/resource_huaweicloud_dli_queue.go]:
```suggestion
		mErr = multierror.Append(mErr, d.Set("scaling_policies", policies))
```
please append setting error to the exist error array

File: huaweicloud/services/dli/resource_huaweicloud_dli_queue.go [hunk 2]
```diff
@@ -297,6 +351,62 @@ func filterByQueueName(body interface{}, queueName string) (r *queues.Queue, err
...
+
+func getQueueScalingPolicies(cfg *config.Config, region, elasticResourcePoolName, queueName string) ([]interface{}, error) {
+	policies, err := getQueuesByName(cfg, region, elasticResourcePoolName, queueName)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(policies) == 0 {
+		return nil, fmt.Errorf("error getting list of the queue scaling policys (%s): %s", queueName, golangsdk.ErrDefault404{})
+	}
```
Comment (Lance52259) [inline huaweicloud/services/dli/resource_huaweicloud_dli_queue.go]: Duplicate codes

File: huaweicloud/services/dli/resource_huaweicloud_dli_queue.go [hunk 3]
```diff
@@ -297,6 +351,62 @@ func filterByQueueName(body interface{}, queueName string) (r *queues.Queue, err
...
+
+	// The `queue_name` parameter supports fuzzy search.
+	for _, queue := range queueList {
+		if queue.QueueName == queueName {
+			return queue.QueueScalingPolicies, nil
+		}
+	}
+
+	return nil, nil
```
Comment (Lance52259) [inline huaweicloud/services/dli/resource_huaweicloud_dli_queue.go]:
```suggestion
	// The `queue_name` parameter supports fuzzy search.
	for _, queue := range queueList {
		if queue.QueueName == queueName {
			return queue.QueueScalingPolicies, nil
		}
	}

	return nil, fmt.Errorf("unable to find the queue (%s) under the elastic resource pool (%s)",
			queueName, elasticResourcePoolName)
```

File: huaweicloud/services/dli/resource_huaweicloud_dli_queue.go [hunk 4]
```diff
@@ -297,6 +351,62 @@ func filterByQueueName(body interface{}, queueName string) (r *queues.Queue, err
...
+
+	if len(queueList) == 0 {
+		return nil, fmt.Errorf("error getting list of the queues (%s) to elastic resource pool (%s): %s",
+			queueName, elasticResourcePoolName, golangsdk.ErrDefault404{})
+	}
+
+	// The `queue_name` parameter supports fuzzy search.
+	for _, queue := range queueList {
+		if queue.QueueName == queueName {
+			return queue.QueueScalingPolicies, nil
```
Comment (Lance52259) [inline huaweicloud/services/dli/resource_huaweicloud_dli_queue.go:372]: return queue object if you want to name this function as the `getQueueByName()` (not getQueuesByName) or rename this function as `getQueueScalingPoliciesByName`

File: huaweicloud/services/dli/resource_huaweicloud_dli_queue.go (hunk 2 above)
Comment (wuzhuanhong) [reply huaweicloud/services/dli/resource_huaweicloud_dli_queue.go]: fixed

File: huaweicloud/services/dli/resource_huaweicloud_dli_queue.go (hunk 1 above)
Comment (wuzhuanhong) [reply huaweicloud/services/dli/resource_huaweicloud_dli_queue.go]: fixed

File: huaweicloud/services/dli/resource_huaweicloud_dli_queue.go (hunk 3 above)
Comment (wuzhuanhong) [reply huaweicloud/services/dli/resource_huaweicloud_dli_queue.go]: fixed

File: huaweicloud/services/dli/resource_huaweicloud_dli_queue.go (hunk 4 above)
Comment (wuzhuanhong) [reply huaweicloud/services/dli/resource_huaweicloud_dli_queue.go:372]: fixed

File: docs/resources/dli_queue.md [hunk 5]
```diff
@@ -91,6 +91,44 @@ The following arguments are supported:
...
+     in which the `priority` value is `1`, the `impact_start_time` value is `00:00`, the `impact_stop_time` value is `24:00`,
+     and the values of `min_cu` and `max_cu` are equal to the number of CUs of the queue.
+     For the default scaling policy, except for the `impact_start_time` and `impact_stop_time`, which are not allowed to
+     be modified, other values can be modified according to the actual situation.
+
+<a name="queue_scaling_policies"></a>
+The `scaling_policies` block supports:
+
+* `priority` - (Required, Int) Specifies the priority of the queue scaling policy.
+  The valid value ranges from `1` to `100`. The larger the value, the higher the priority.
```
Comment (Lance52259) [inline docs/resources/dli_queue.md]:
```suggestion
  The valid value ranges from `1` to `100`. The larger value means the higher priority.
```

File: docs/resources/dli_queue.md [hunk 6]
```diff
@@ -91,6 +91,44 @@ The following arguments are supported:
...
+
+* `min_cu` - (Required, Int) Specifies the minimum number of CUs allowed by the scaling policy.
+  The number must be a multiple of `4`.
+
+  -> The total minimum CUs of all queues in an elastic resource pool cannot be more than the minimum CUs of the pool.
+
+* `max_cu` - (Required, Int) Specifies the maximum number of CUs allowed by the scaling policy.
+  The number must be a multiple of `4`.
+  
+  ->The maximum CUs of any queue in an elastic resource pool cannot be more than the maximum CUs of the pool.
```
Comment (Lance52259) [inline docs/resources/dli_queue.md]:
```suggestion
  -> The maximum CUs of any queue in an elastic resource pool cannot be more than the maximum CUs of the pool.
```

Comment (wuzhuanhong) [reply docs/resources/dli_queue.md]: fixed

File: docs/resources/dli_queue.md (hunk 5 above)
Comment (wuzhuanhong) [reply docs/resources/dli_queue.md]: fixed

**Pull request merged**